cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_controller.go

```diff
@@ -285,15 +299,16 @@ func newMachineController(
 	kubeInformerFactory := kubeinformers.NewSharedInformerFactory(kubeclient, 0)
 	informerFactory := dynamicinformer.NewFilteredDynamicSharedInformerFactory(dynamicclient, 0, metav1.NamespaceAll, nil)
 
-	// TODO(alberto): let environment variable to override defaultMachineAPI
-	machineDeploymentResource, _ := schema.ParseResourceArg(fmt.Sprintf("machinedeployments.%v", defaultMachineAPI))
+	specifiedMachineAPI := getAPIVersionGroup()
```

Comment:
This seems like something you would pass in as a parameter to newMachineController but maybe it's more practical here so we have to carry patches on fewer files.

Reply:
yeah, i had thought about that but i wasn't sure how deep to cut here. it feels a little weird pulling an env variable at this point, i am happy to change it to a parameter if that makes more sense.

Reply:
>This seems like something you would pass in as a parameter to newMachineController but maybe it's more practical here so we have to carry patches on fewer files

yeh this overridability is only useful if it saves us from carrying commits which we can only get to with the env variable

Reply:
just to be clear, is it ok to leave the env variable parsing here or should i change it to a parameter of `newMachineController` ?

Reply:
Env is good because that will enable us reuse this upstream. Afaik providers can not expose extra parameters so we could only expose then in core and carry over a patch for it downstream which would defeat the purpose for making this overridable.

Reply:
ack, i didn't think we would add a new command line flag, but if i understood @michaelgugino i think i could move the environment variable parsing to the code that calls `newMachineController`. i'm just not sure if that's the best choice given all the reflected changes that will need to be made.